### PR TITLE
Added MailboxHandler

### PR DESCRIPTION
**Pull request by aducret**

### CODE REVIEW

**Comment by pablog** [comment]
What do you think of using names `MessageProducer` (`UIControl`) and `MessageForwarder` (`UIView`)? @guidomb

File: Portal/View/UIKit/Renderers/MessageHandler.swift [hunk 1]
```diff
@@ -9,34 +9,25 @@
...
-    func getDispatcher<MessageType>(for event: UIControlEvents) -> MessageDispatcher<MessageType>?
-    
     func register<MessageType>(dispatcher: MessageDispatcher<MessageType>, for event: UIControlEvents)
     
     func unregisterDispatcher<MessageType>(for event: UIControlEvents) -> MessageDispatcher<MessageType>?
     
 }
 
-extension MessageHandler {
+protocol MailBoxHandler where Self: UIView {
```
Comment (pablog) [inline Portal/View/UIKit/Renderers/MessageHandler.swift]: I'd move this protocol above the other protocol to make it easier to read.

**Comment by guidomb** [comment]
Can somebody explain why do we need this?

**Comment by pablog** [comment]
Because current protocol `MessageHandler` applies to `UIControl`.

So a component as `UIView` which doesn't inherit from `UIControl` could not implement `MessageHandler`, but it needs to have a mailbox to forward the messages from its sub components.

So, there will be 2 protocols, one used in those components that can create and send messages, and another one used in those components that will only forward messages from their sub components.

We initially thought we'd need this change in our current components `Label` and `TextView` which their corresponding `UIKit` components don't inherit `UIControl`, but given we agreed these will always be used as leaves, we don't need it right now. Anyway, when we implement the "diffability" for `Container` it will be necessary.

**Comment by guidomb** [comment]
What do you guys think about the following renames

`MessageHandler` => `UIEventSource`

`MailboxHandler` => `MessageEmitter`

I like `MessageProducer` but I am not convinced about `MessageForwarder` but I don't have strong arguments. I think either solution would be better than what we currently have.

**Comment by pablog** [comment]
I don't like that for 3 reasons:

- I'd be consistent by using either `event` or `message` in both.
- I'd use `UI` in both or none.
- Emitter sounds like if it generates the message, instead of only taking other messages and passing them up in the tree.

**Comment by guidomb** [comment]
Yeah makes sense. Lets go with what you proposed @pablog

**Review by pablog — APPROVED** [review]

**Pull request merged**